internal/garbage/freight_test.go

```diff
@@ -0,0 +1,288 @@
+package garbage
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+)
+
+func TestCleanProjectFreight(t *testing.T) {
+	testCases := []struct {
+		name       string
+		collector  *collector
+		assertions func(error)
```

Comment:
I've been aware of the pattern, but personally never minded it aborting the outer test. I think I saw "fail faster" as reasonable in many cases, but I don't feel strongly about it. Happy to start including the t arg.

Reply:
The reason why I personally like it to be contained is because it provides a better overview in a single run of the things it fails on, instead of potentially addressing one issue and then having the next thing tripping over.

Fast failing behavior can still be achieved by using `go test -failfast`.

Reply:
Good point. I am on board.